### DISCUSSION TITLE
Allow or document how invoke Lambda function locally via Bref

### DISCUSSION THREAD

**Discussion by grasmash** [brefphp/bref#1283] [Support]
I currently use bref for a Lambda function that is exposed via a Function URL. I can test my function locally by running a local php server and sending a Curl request directly to my handler script. However, this bypasses all of bref's response handling. 

Is it possible to locally invoke the Lambda function in a way that routes the request through bref?

**Comment by afu-dev** [comment]
Bref has two pages of documentation regarding local development:

- webapp: https://bref.sh/docs/web-apps/local-development.html
- functions: https://bref.sh/docs/function/local-development.html

You can also checkout the Bref dev-server repo: https://github.com/brefphp/dev-server

Does this help you?

**Comment by grasmash** [comment]
I've read through both of those docs, but they only offer how to test 1) event driven invokes, and 2) direct http requests to the handler (bypassing bref). 

I've tried this running spinning up Docker and running this in a test:
```php
        $event = json_decode(file_get_contents(__DIR__ . '/fixtures/request.json'), TRUE);
        $handler = new FpmHandler('index.php');
        $handler->start();
        $invoker = new Invoker;
        $context = new Context('test', 0, '', '');
        $result = $invoker->invoke($handler, $event, $context);
        $handler->stop();
```

However,  I get:
```
[23-Sep-2022 17:50:40] NOTICE: Failed implicitly binding to ::, retrying with 0.0.0.0
[23-Sep-2022 17:50:40] ERROR: unable to bind listening socket for address '9001': Address already in use (98)
[23-Sep-2022 17:50:40] ERROR: FPM initialization failed
```

Because FPM is already running in the container. For some reason I can't seem to kill the process.

**Reply by mnapoli** [reply]
What about "the simple way" documented here: https://bref.sh/docs/web-apps/local-development.html#the-simple-way

Any reason it wouldn't work for you?
